### PR TITLE
Assignment8

### PR DESCRIPTION
**Pull request by prajjwalkumarpanzade**
Migration and Models file

### CODE REVIEW

File: Assignment8/models/book.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Book < ApplicationRecord
+  validates :title, presence: true , uniqueness: true
```
Comment (samu80) [inline Assignment8/models/book.rb]: comma should be present just after true as:
 `validates :title, presence: true,  uniqueness: true`

File: Assignment8/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :phone_no, presence: true, uniqueness: true, format: { with: /\A[0-9]+\z/, message: "Integer only." } 
+  validates :role, presence: true, inclusion: { in: %w(user teacher librarian), message: "%{value} is not a valid role" }
+  before_validation :normalize_role,  on: :create
```
Comment (samu80) [inline Assignment8/models/user.rb:6]: Add line space here

File: Assignment8/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :phone_no, presence: true, uniqueness: true, format: { with: /\A[0-9]+\z/, message: "Integer only." } 
+  validates :role, presence: true, inclusion: { in: %w(user teacher librarian), message: "%{value} is not a valid role" }
+  before_validation :normalize_role,  on: :create 
+  after_save :welcome
+  has_many :transactions
+  private
```
Comment (samu80) [inline Assignment8/models/user.rb:10]: Add line space here

File: Assignment8/models/book.rb (hunk 1 above)
Comment (prajjwalkumarpanzade) [reply Assignment8/models/book.rb]: ok

File: Assignment8/models/user.rb (hunk 3 above)
Comment (prajjwalkumarpanzade) [reply Assignment8/models/user.rb:10]: done

File: Assignment8/models/user.rb (hunk 2 above)
Comment (prajjwalkumarpanzade) [reply Assignment8/models/user.rb:6]: done

**Review by samu80 — APPROVED** [review]

**Review by samu80 — APPROVED** [review]

**Pull request merged**